src/network/jsonwebtask.cpp

```diff
@@ -244,12 +244,24 @@ QNetworkReply* JsonWebTask::doStartNetworkRequest(
     DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
     DEBUG_ASSERT(networkAccessManager);
 
-    DEBUG_ASSERT(m_baseUrl.isValid());
+    VERIFY_OR_DEBUG_ASSERT(m_baseUrl.isValid()) {
+        kLogger.warning() << "Invalid base URL" << m_baseUrl;
```

Comment:
This will produce two log entries: 
Warning: Invalid base URL
Warning: Network task has not been started

But the caller TagFetcher::slotFingerprintReady is not informed. The timeout is also not used. 
I am afraid this produces an inconsistent state now. 

I think WebTask::slotStart requires an addition onNetworkError call.

Reply:
The two log messages cannot be combined. They refer to different aspects, namely the cause followed by the consequences. The second is even only a debug log, not a warning.

I'll check if we need another signal and a final `NotStarted` state here.

Reply:
This can be handled by setting the state to `Aborted` and sending the corresponding signal (without a request URL). Good catch, another dead end of the state machine fixed.